framework/rpc/grpc.py

```diff
@@ -95,10 +96,30 @@ def __init__(self, rpc_host):
         # Cache gRPC channels per port
         self.channels = dict()
 
-    def _make_channel(self, port) -> grpc.Channel:
+    def _make_channel(self, port, secure_channel=False) -> grpc.Channel:
         if port not in self.channels:
             target = f"{self.rpc_host}:{port}"
-            self.channels[port] = grpc.insecure_channel(target)
+            if secure_channel:
+                result = subprocess.run(
+                    [
+                        "gcloud",
+                        "auth",
+                        "print-identity-token",
+                    ],
+                    check=True,
+                    capture_output=True,
```

Comment:
If we go with this route, `timeout` is a must. IRRC, otherwise it'll be allowed to hang forever. Also, I'd like some logging before the command to indicate what's running.